### PR TITLE
Option class for consensus timeout configuration

### PR DESCRIPTION
**Pull request by colibrishin**
This PR introduces a new class for the option of consensus timeout configuration.

## Base context
- For each step (Propose, PreVote, PreCommit), there are base timeouts and multipliers. Base timeouts are the initial values for timeout, and multipliers are for multiplying factors when a new round is started.
- This class guards from unexpected values (where value <= 0) and lets the values can be configurable.

### CODE REVIEW

File: Libplanet.Net/Consensus/ContextTimeoutOption.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(
+                        nameof(value),
+                        "ProposeSecondBase must be greater than or equal to 0."
+                    );
+                }
+
+                _proposeSecondBase = value;
+            }
```
Comment (greymistcube) [inline Libplanet.Net/Consensus/ContextTimeoutOption.cs:31]: Might be just my personal preference, but I'm not a big fan of setters where mutation doesn't make much sense. 🙄
Is there a context in which we would want to change the option on the fly? If not, I'd prefer using default values for a constructor.

**Review by greymistcube — APPROVED** [review]

File: Libplanet.Net/Consensus/ContextTimeoutOption.cs (hunk 1 above)
Comment (colibrishin) [reply Libplanet.Net/Consensus/ContextTimeoutOption.cs:31]: Hmm, I also don't know if there's any such use-case, and furthermore, if it should be permitted.

Comment (colibrishin) [reply Libplanet.Net/Consensus/ContextTimeoutOption.cs]: Changed to the constructor style.
https://github.com/planetarium/libplanet/commit/9568ed60af44d865baaa9ba196700cdbb5398253#diff-95065cc3ba907122e5352fa2ee987c580d32bfe93d8c8e5a71b70a23e2767aeb

File: Libplanet.Net/Consensus/ContextTimeoutOption.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        public int PreVoteSecondBase { get; } = 5;
+
+        public int PreCommitSecondBase { get; } = 5;
+
+        public int ProposeMultiplier { get; } = 1;
+
+        public int PreVoteMultiplier { get; } = 1;
+
+        public int PreCommitMultiplier { get; } = 1;
```
Comment (greymistcube) [inline Libplanet.Net/Consensus/ContextTimeoutOption.cs:86]: I think you can remove default assignments here as parameter-less default constructor is not available. 🤔

Comment (colibrishin) [reply Libplanet.Net/Consensus/ContextTimeoutOption.cs]: I see. One thing I noticed is that struct default sets all values to zero, but class default treats the same as null.
https://github.com/planetarium/libplanet/pull/2382/commits/7fa12b6a7634b75b2dae35b83a303e094d1b0cf7#diff-95065cc3ba907122e5352fa2ee987c580d32bfe93d8c8e5a71b70a23e2767aeb

**Review by greymistcube — APPROVED** [review]

**Pull request merged**